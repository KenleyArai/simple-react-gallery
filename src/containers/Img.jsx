import React, { Component } from "react";
import PropTypes from "prop-types";

import ImgStyle from "ImgStyle";

export default class Img extends Component {
  static propTypes = {
    src: PropTypes.string,
    srcSet: PropTypes.string,
    theme: PropTypes.shape({
      primary: PropTypes.string,
      secondary: PropTypes.string,
      tertiary: PropTypes.string
    })
  };

  static defaultProps = {
    src: "",
    srcSet: "",
    theme: {
      primary: "",
      secondary: "",
      tertiary: ""
    }
  };

  render() {
    var { src, srcSet, theme } = { ...this.props };
    if (src) return <ImgStyle src={src} />;
    return <ImgStyle srcSet={srcSet} />;
  }
}
