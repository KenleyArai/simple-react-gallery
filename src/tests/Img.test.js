import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";
configure({ adapter: new Adapter() });

import Img from "containers/Img";
import testSrc from "../logo-small.svg";

describe("Img", () => {
  it("should render without", () => {
    const wrapper = shallow(<Img />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with src", () => {
    const wrapper = shallow(<Img srcSet={testSrc} />);
    expect(wrapper).toMatchSnapshot();
  });
});
