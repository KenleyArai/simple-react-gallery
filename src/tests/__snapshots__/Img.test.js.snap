// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img should render with src 1`] = `
<styled.img
  src="logo-small.svg"
/>
`;

exports[`Img should render with srcSet 1`] = `
<styled.img
  srcSet="logo-small.svg"
/>
`;

exports[`Img should render without src 1`] = `
<styled.img
  srcSet=""
/>
`;
